### PR TITLE
test: add more tests to topology spread constraints plugin

### PR DESCRIPTION
**Pull request by michaelawyu**
### Description of your changes

This PR adds more tests to the topology spread constraints plugin.

I have:

- [x] Run `make reviewable` to ensure this PR is ready for review.

### How has this code been tested

N/A

### Special notes for your reviewer

A minor part of this PR overlaps with another test PR.

### CODE REVIEW

File: pkg/scheduler/framework/plugins/topologyspreadconstraints/utils_test.go [hunk 1]
```diff
@@ -949,3 +950,1083 @@ func TestWillViolate(t *testing.T) {
...
+		},
+	}
+
+	for _, tc := range testCases {
+		t.Run(tc.name, func(t *testing.T) {
+			state := framework.NewCycleState(tc.clusters, tc.bindings)
+
+			violations, scores, err := evaluateAllConstraints(state, tc.doNotSchedule, tc.scheduleAnyway)
+			if err != nil {
+				t.Errorf("evaluateAllConstraints() = %v, want no error", err)
```
Comment (zhiying-lin) [inline pkg/scheduler/framework/plugins/topologyspreadconstraints/utils_test.go:1609]:
```suggestion
				t.Fatalf("evaluateAllConstraints() = %v, want no error", err)
```

not need to compare the result?

File: pkg/scheduler/framework/plugins/topologyspreadconstraints/utils_test.go [hunk 2]
```diff
@@ -949,3 +950,1083 @@ func TestWillViolate(t *testing.T) {
...
+			},
+		},
+	}
+
+	for _, tc := range testCases {
+		t.Run(tc.name, func(t *testing.T) {
+			state := framework.NewCycleState(tc.clusters, tc.bindings)
+			genPluginState, err := prepareTopologySpreadConstraintsPluginState(state, tc.policy)
+			if err != nil {
+				t.Errorf("prepareTopologySpreadConstraintsPluginState() = %v, want no error", err)
```
Comment (zhiying-lin) [inline pkg/scheduler/framework/plugins/topologyspreadconstraints/utils_test.go:2024]: ditto

**Review by zhiying-lin — APPROVED** [review]
two minor comments, others LTGM

File: pkg/scheduler/framework/plugins/topologyspreadconstraints/utils_test.go (hunk 1 above)
Comment (michaelawyu) [reply pkg/scheduler/framework/plugins/topologyspreadconstraints/utils_test.go:1609]: Done.

File: pkg/scheduler/framework/plugins/topologyspreadconstraints/utils_test.go (hunk 2 above)
Comment (michaelawyu) [reply pkg/scheduler/framework/plugins/topologyspreadconstraints/utils_test.go:2024]: Done.

**Comment by michaelawyu** [comment]
GitHub is acting a little weird; the latest commit is not picked up. Will retry tomorrow.

**Review by zhiying-lin — APPROVED** [review]

**Comment by michaelawyu** [comment]
Thanks, @Arvindthiru!

**Pull request merged**